hubblestack/filter/filter_chain.py

```diff
@@ -0,0 +1,83 @@
+##
+
+import importlib
+import logging
+import yaml
+
+import hubblestack.filter.seq_id as seq_id
+from hubblestack.exceptions import CommandExecutionError
+import hubblestack.modules.cp
+
+log = logging.getLogger(__name__)
+
+chains = {}
+
+class FilterChain:
+    """
+    FilterChain loads the filter config from the hubble profile filter_chain.yaml
+    This configuraiton file will have a default configuration, as well as options for
+    overriding with returner specific filter order
+    example:
+
+    default:
+      sequence:   # Label
+        type: "hubblestack.filter.seq_id" # filter type
+        label: "seq" # optional: what field name to add
+        padding: 10 # seq: 0000000001, 0000000002
+    """
+
+    def __init__(self, config_path, config_label="default"):
+        """
+        config_path - path to yaml file defining the configuration for the filter chain
+        config_label - the label in the yaml file underwhich the filters and their configurations are located
+        """
+        self.config_path = config_path
+        self.config_label = config_label
+        self.config = {}
+        self.chain = []   
+        self._load_config()
+
+    def _load_config(self):
+        config_path = "salt://filter_chain.yaml"
+        config_path = __mods__["cp.cache_file"](config_path)
+        try:
+            with open(config_path, 'r') as handle:
+                self.config = yaml.safe_load(handle)
+        except Exception as e:
+            self.config = {"default": {"filter": { "default": {
+                 "sequence_id": { "label": "seq", "type": "hubblestack.filter.seq_id" },
+                 "hubble_version": { "label": "hubble_version", "type": "hubblestack.hubble_version"}}}}}
+            raise CommandExecutionError(f"Could not load filter config: {e}")
+
+        if not isinstance(self.config, dict) or \
+            "filter" not in self.config or \
+            not(isinstance(self.config["filter"], dict)) or \
+            self.config_label not in self.config["filter"].keys() or \
+            not(isinstance(self.config["filter"][self.config_label], dict)):
+            raise CommandExecutionError("FilterChain config not formatted correctly")
+
+        self.config = self.config['filter'][self.config_label]
+ 
+        for filter_name in self.config:
+            new_fltr = self._get_filter_class(self.config[filter_name]["type"])(filter_name, self.config[filter_tag])
+            self.chain.append(new_fltr)
```

Comment:
Follow-up on my `property` suggestion. It was missing the `return`, as @mzainea said, and now `config` can be used as a variable inside the method's scope. Nevertheless, when we do `FilterChain.config` in some other code, it will return the config proper.

```suggestion
        ```suggestion
        self._config = self.config
        self.chain = []

    @property
    def config(self):
        config_path = "salt://filter_chain.yaml"
        config_path = __mods__["cp.cache_file"](config_path)
        try:
            with open(config_path, 'r') as handle:
                config = yaml.safe_load(handle)
        except Exception as e:
                config = {"default": {"filter": { "default": {
                 "sequence_id": { "label": "seq", "type": "hubblestack.filter.seq_id" },
                 "hubble_version": { "label": "hubble_version", "type": "hubblestack.hubble_version"}}}}}
            raise CommandExecutionError(f"Could not load filter config: {e}")

        if not isinstance(config, dict) or \
            "filter" not in config or \
            not(isinstance(config.get("filter"), dict)) or \
            config_label not in config["filter"].keys() or \
            not(isinstance(config.get("filter", {}).get(self.config_label), dict)):
            raise CommandExecutionError("FilterChain config not formatted correctly")

        config = config['filter'][config_label]
        
        for filter_name in self.config:
            new_fltr = self._get_filter_class(config[filter_name]["type"])(filter_name, config[filter_tag])
            self.chain.append(new_fltr)

         return config
```